packages/cli/src/internal/prompt/text.ts

```diff
@@ -264,7 +265,18 @@ function basePrompt(
     validate: Effect.succeed,
     ...options
   }
-  return InternalPrompt.custom(initialState, {
+
+  const state: State = {
+    ...initialState,
+    ...(Predicate.isString(options.default) ?
+      {
+        value: options.default,
+        cursor: options.default.length
+      } :
+      undefined)
+  }
```

Comment:
@cdierkens - I don't think we actually need to modify the initial state here.

I think the behavior we are looking for is to have the `default` value be used in place of the `state.value` if `state.value` is empty when the user actually submits the prompt.

Since we have access to the `default` value in the `opts` that are passed to the prompt processing functions, I think we can just make use of that instead.

I think the only change we need to make is in `handleProcess`, to use the `default` if `state.value` is empty:

```diff
      case "enter":
      case "return": {
+        const value = state.value.length > 0 ? state.value : options.default
-        return Effect.match(options.validate(state.value), {
+        return Effect.match(options.validate(value), {
          onFailure: (error) =>
            Action.NextFrame({
              state: { ...state, error: Option.some(error) }
            }),
          onSuccess: (value) => Action.Submit({ value })
        })
      }
```

Reply:
I've updated with your suggestion.  It's a nice bite size and testable increment of value. 

I was hoping to follow up with another PR to render the default value during the prompt. Here's an example from `create-remix`

https://github.com/user-attachments/assets/5c224f4a-5026-4acb-a3bd-8971b15bfa64

What's happening here is that the initial state is empty, so the default is rendered in a grayed out manner. You can tab to accept the default value. Once you delete the state value again, you are re-presented with the grayed out default.

I can follow up with an update to `renderInput` where the default value is rendered in gray when the state's value is empty. Let me know your thoughts!

Reply:
That sounds good to me - I can help with the finer points of that PR as well if you like 👍